hide hidden fields even if they have a value

machisuji opened the pull request:
Addendum to WP [#2294](https://community.openproject.com/work_packages/22994/activity)

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://feature-wp-config-hidden.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://feature-wp-config-hidden.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

ulferts commented:
@machisuji please rebase and check the tests.

oliverguenther committed:
Make test/dev assets settings explit

machisuji committed:
description must not be customizable

machisuji committed:
updated feature help text

machisuji committed:
fields with default values can be hidden

machisuji committed:
fixed hide condition

machisuji committed:
fixed edit wp spec (required cf)

for some reason factories initialize custom fields
with '' (empty string) as default values.
The API interprets this as a default value being present.
In normal operation a custom field without a default value
will return nil, not the empty string
